.github/workflows/codeql-analysis.yml

```diff
@@ -49,7 +49,7 @@ jobs:
 
     # Install a more recent Go that understands modern go.mod content.
     - name: Install Go
-      uses: actions/setup-go@v4
+      uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
```

Comment:
Breaking change from 4.x -> 5.x is the Node 20 requirement: https://github.com/actions/setup-go/releases/tag/v5.0.0